storage/innobase/fil/fil0crypt.cc

```diff
@@ -3144,3 +3145,169 @@ fil_space_verify_crypt_checksum(
 
 	return(encrypted);
 }
+
+redo_log_key* redo_log_keys::load_latest_key() {
+	size_t   klen = 0;
+	size_t   klen2 = 0;
+	uint    version = 0;
+	char *key_type = NULL;
+	byte *rkey = NULL;
+	byte *rkey2 = NULL;
+
+	if (my_key_fetch(PERCONA_REDO_KEY_NAME, &key_type, NULL,
+		reinterpret_cast<void**>(&rkey), &klen) || rkey == NULL ||
+		strncmp(key_type, "AES", 3) != 0) {
```

Comment:
If only 3 chars are compared, this does prefix comparison only?
```suggestion
		strncmp(key_type, "AES", 4) != 0) {
```